File: server/datasets/java_source-code/Multiple-A3/organic_5.java
// Linked List Reversal - organic submission 5
// Iterative with length() helper and different variable names.

class Node {
    int data;
    Node next;

    public Node(int data) {
        this.data = data;
        this.next = null;
    }
}

class LinkedList {
    Node head;

    public LinkedList() {
        this.head = null;
    }

    public void pushBack(int val) {
        Node n = new Node(val);
        if (this.head == null) {
            this.head = n;
            return;
        }
        Node ptr = this.head;
        while (ptr.next != null) {
            ptr = ptr.next;
        }
        ptr.next = n;
    }

    public int length() {
        int count = 0;
        Node cur = this.head;
        while (cur != null) {
            count++;
            cur = cur.next;
        }
        return count;
    }

    public void invert() {
        Node before = null;
        Node target = this.head;
        while (target != null) {
            Node after = target.next;
            target.next = before;
            before = target;
            target = after;
        }
        this.head = before;
    }

    public void show() {
        Node c = this.head;
        StringBuilder sb = new StringBuilder();
        while (c != null) {
            sb.append(c.data);
            if (c.next != null) sb.append(" -> ");
            c = c.next;
        }
        System.out.println(sb.toString());
    }

    public static void main(String[] args) {
        LinkedList ll = new LinkedList();
        for (int i = 1; i <= 6; i++) ll.pushBack(i * 10);
        System.out.println("len=" + ll.length());
        ll.show();
        ll.invert();
        ll.show();
    }
}

File: server/datasets/controlled_benchmarks/python/multi_student_cohorts/cohort_a3/organic_5.py
# Linked List Reversal - organic submission 5
# Iterative, but with a length() helper and totally different variable names.

class Node:
    def __init__(self, data):
        self.data = data
        self.next = None


class LinkedList:
    def __init__(self):
        self.head = None

    def push(self, data):
        node = Node(data)
        if self.head is None:
            self.head = node
            return
        tail = self.head
        while tail.next is not None:
            tail = tail.next
        tail.next = node

    def length(self):
        count = 0
        node = self.head
        while node is not None:
            count += 1
            node = node.next
        return count

    def reverse(self):
        node_before = None
        node_at = self.head
        while node_at is not None:
            node_after = node_at.next
            node_at.next = node_before
            node_before = node_at
            node_at = node_after
        self.head = node_before

    def __str__(self):
        vals = []
        node = self.head
        while node is not None:
            vals.append(repr(node.data))
            node = node.next
        return "[" + ", ".join(vals) + "]"


if __name__ == "__main__":
    ll = LinkedList()
    for v in ["a", "b", "c", "d"]:
        ll.push(v)
    print("len:", ll.length())
    print(ll)
    ll.reverse()
    print(ll)
